File: TypeScript/test-project/basic/person.ts
interface Person { 
	readonly id: number; // reaondly 
	name: string;
	age?: number;  // ? optional 
	[propName: number]: string; // any property
}

(()=>{
	let c: Person = { id: 1, name: 'jialong', 1: 'one', 3: 'three' }
	let d: Person = { id: 2, name: 'yabe', age: 8 }

	let p: Person[] = [c, d]
	let p_t: Array<Person> = [d,c] // Array Generic 

	console.log(p)
	console.log(p_t)
})()

File: Lua/person.lua

local Person = {}

Person.__index = Person

function Person:new(name, age)
    local instance = setmetatable({}, Person)
    instance.name = name
    instance.age = age
    return instance
end


function Person:info()
    print("Name: " .. self.name .. "\nAge: " .. self.age)
end

local p = Person:new("yabe", 1035)

p:info()








